### PR TITLE
boards: st_stm32: Clean up driver generic flags from boards' _defconfig files

### PR DESCRIPTION
**Pull request by ydamigos**
This is the first step for unifying STM32 configuration files as mentioned in #3483.

It removes series specific CONFIG flags from boards' _defconfig files.

### CODE REVIEW

File: boards/arm/96b_carbon/Kconfig.defconfig
Comment (mbolivar) [inline boards/arm/96b_carbon/Kconfig.defconfig]: Isn't this more of an application specific setting than a board setting?

File: boards/arm/96b_neonkey/Kconfig.defconfig
Comment (mbolivar) [inline boards/arm/96b_neonkey/Kconfig.defconfig]: Same here, and elsewhere -- IRQ priorities are up to the application, not the board, no? There's nothing board-specific about this I can see.

File: boards/arm/nucleo_f429zi/nucleo_f429zi_defconfig
Comment (mbolivar) [inline boards/arm/nucleo_f429zi/nucleo_f429zi_defconfig]: The comment says port G is also enabled, but that's not done here. Which should it be?

**Review by mbolivar — COMMENTED** [review]
This is really nice, thorough work.

I looked at the common file and the F4 changes and this all looks sane, with a couple of nits.

File: boards/arm/96b_carbon/Kconfig.defconfig
Comment (ydamigos) [reply boards/arm/96b_carbon/Kconfig.defconfig]: It is, but we already define I2C and UART IRQ priority in SOC's dtsi files. Moreover, we specify a default SPI_*_IRQ_PRI value for other boards in the same way (e.g. hexiwear_k64, em_starterkit). We can override these default values at application level using a dt overlay file or prj.conf.

File: boards/arm/96b_neonkey/Kconfig.defconfig
Comment (ydamigos) [reply boards/arm/96b_neonkey/Kconfig.defconfig]: Same as above.

File: boards/arm/nucleo_f429zi/nucleo_f429zi_defconfig
Comment (ydamigos) [reply boards/arm/nucleo_f429zi/nucleo_f429zi_defconfig]: It was not enabled before this patch. I checked boards pinmux file and port G is used for the ethernet, so I will also enable it.

File: boards/arm/96b_carbon/Kconfig.defconfig
Comment (mbolivar) [reply boards/arm/96b_carbon/Kconfig.defconfig]: I didn't know about that; thanks for the pointers. I still think board level is the wrong place to put this, though.

I could maybe see arguments to do it at SoC level (though on ARM, I think using the default priority ordering determined by the NVIC makes more sense).

Do you know of a design rationale for why it's done elsewhere? I'm not sure it makes sense to copy what's done elsewhere without good reason here.

Comment (ydamigos) [reply boards/arm/96b_carbon/Kconfig.defconfig]: I just did it to follow the same approach as other boards, until I push a PR for STM32 devices to get SPI configuration from DT (like I2C and USB).

Comment (mbolivar) [reply boards/arm/96b_carbon/Kconfig.defconfig]: IMO it'd be better to drop this from the board files. Otherwise when the values in board files start to drift, it could be a source of confusing bugs when porting an application between boards. But I don't think this should block the merge since it can be overridden.

**Comment by ydamigos** [comment]
Is there a way to remove reviewers? A rebase from arm branch to master branch requested review from people who may not be interested.

**Comment by pfalcon** [comment]
@ydamigos : You can, if you can add reviewers. If not, let me know whom more you'd like to review. Perhaps, avoid "@" syntax to not disturb those folks, just raw nicks.

**Comment by ydamigos** [comment]
@pfalcon Thanks. Could you also remove andrewboie, andyross, dbkinder, ruuddw, sjanc?

**Comment by pfalcon** [comment]
done

File: boards/arm/96b_carbon/Kconfig.defconfig
Comment (ydamigos) [reply boards/arm/96b_carbon/Kconfig.defconfig]: I removed SPI_*_IRQ_PRI (I left it only on carbon because I wanted to run the SPI test cases on it) from the board files. I used test case filtering to disable some tests when the peripheral is not supported. I also push a PR to enable STM32 SPI devices to get their configuration from DT.

File: boards/arm/96b_carbon/96b_carbon_defconfig
Comment (erwango) [inline boards/arm/96b_carbon/96b_carbon_defconfig]: I'm not sure about GPIO activation. Can we move it under arch and "if GPIO"
Drawback is that it will activate all GPIO port by default, but might be less error prone for new comers.
And footprint optimisation can still happen in this file.

File: boards/arm/nucleo_f030r8/nucleo_f030r8_defconfig
Comment (erwango) [inline boards/arm/nucleo_f030r8/nucleo_f030r8_defconfig]: Clean this one

File: boards/arm/nucleo_f030r8/nucleo_f030r8_defconfig
Comment (erwango) [inline boards/arm/nucleo_f030r8/nucleo_f030r8_defconfig]: Since this is a board default configuration, can UART_2 activation be moved to Kconfig.defconfig under condition `if UART_CONSOLE`

File: boards/arm/nucleo_f091rc/nucleo_f091rc_defconfig
Comment (erwango) [inline boards/arm/nucleo_f091rc/nucleo_f091rc_defconfig]: Could be cleaned

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks for this work!
Some comments. Main points are about GPIO and configuration of board default HW (BSP).

File: boards/arm/disco_l475_iot1/disco_l475_iot1_defconfig
Comment (mbolivar) [inline boards/arm/disco_l475_iot1/disco_l475_iot1_defconfig]: Thanks for the update! I'm confused though: IRQ_PRI is still in this board file, but you said only Carbon should be left. Did you get them all?

Comment (ydamigos) [reply boards/arm/disco_l475_iot1/disco_l475_iot1_defconfig]: I am sorry, I mentioned the one I added. This was already there. I remove them both in the PR which enables STM32 SPI devices to get their configuration from DT.

Comment (mbolivar) [reply boards/arm/disco_l475_iot1/disco_l475_iot1_defconfig]: Ah, okay, thanks! That makes sense.

File: boards/arm/96b_carbon/96b_carbon_defconfig
Comment (ydamigos) [reply boards/arm/96b_carbon/96b_carbon_defconfig]: I moved GPIO definition for ports A, B, C in common Kconfig.defconfig.series and enable the rest ports per SOC in SOC's Kconfig.defconfig.series or Kconfig.defconfig.stm32*.

File: boards/arm/nucleo_f030r8/nucleo_f030r8_defconfig
Comment (ydamigos) [reply boards/arm/nucleo_f030r8/nucleo_f030r8_defconfig]: Done.

File: boards/arm/nucleo_f030r8/nucleo_f030r8_defconfig
Comment (ydamigos) [reply boards/arm/nucleo_f030r8/nucleo_f030r8_defconfig]: Done.

File: boards/arm/nucleo_f091rc/nucleo_f091rc_defconfig
Comment (ydamigos) [reply boards/arm/nucleo_f091rc/nucleo_f091rc_defconfig]: Done.

File: boards/arm/stm32373c_eval/stm32373c_eval_defconfig
Comment (ydamigos) [reply boards/arm/stm32373c_eval/stm32373c_eval_defconfig]: Done.

File: boards/arm/stm32373c_eval/stm32373c_eval_defconfig
Comment (ydamigos) [reply boards/arm/stm32373c_eval/stm32373c_eval_defconfig]: Done.

**Comment by galak** [comment]
@erwango you good with this now?

File: arch/arm/soc/st_stm32/stm32f4/Kconfig.defconfig.series
Comment (erwango) [inline arch/arm/soc/st_stm32/stm32f4/Kconfig.defconfig.series]: Why not adding E and H here?
From quick search they are supported on all available SoCs

File: arch/arm/soc/st_stm32/stm32f4/Kconfig.defconfig.stm32f429xx
Comment (erwango) [inline arch/arm/soc/st_stm32/stm32f4/Kconfig.defconfig.stm32f429xx]: E to K supported

File: boards/arm/nucleo_f030r8/Kconfig.defconfig
Comment (erwango) [inline boards/arm/nucleo_f030r8/Kconfig.defconfig]: Is that due to memory footprint issue?
In any case can you add little bit more information so we can know the exact issue?

**Review by erwango — CHANGES_REQUESTED** [review]
Really great!

One general comment on GPIOs: you made the choice to enable GPIO ports depending on what was enabled, in each board, before this series.
While I understand this choice, when we move GPIO ports activation in SoC, it makes choice of active GPIOs quite arbitrary.
Since it is not a board choice anymore, can you activate all GPIOs ports rather than the current arbitrary selection?

File: arch/arm/soc/st_stm32/stm32f4/Kconfig.defconfig.series
Comment (ydamigos) [reply arch/arm/soc/st_stm32/stm32f4/Kconfig.defconfig.series]: Fixed.

File: arch/arm/soc/st_stm32/stm32f4/Kconfig.defconfig.stm32f429xx
Comment (ydamigos) [reply arch/arm/soc/st_stm32/stm32f4/Kconfig.defconfig.stm32f429xx]: Fixed.

File: boards/arm/nucleo_f030r8/Kconfig.defconfig
Comment (ydamigos) [reply boards/arm/nucleo_f030r8/Kconfig.defconfig]: Updated.

**Review by erwango — APPROVED** [review]
Once again, thanks for this work. Let's get it merged now

**Comment by ydamigos** [comment]
I pushed an update to just fix some commits with empty bodies after the decision in yesterday's TSC meeting.

**Comment by erwango** [comment]
> I pushed an update to just fix some commits with empty bodies after the decision in yesterday's TSC meeting.

I had missed that. Thanks for pointing it.

**Comment by erwango** [comment]
@galak : Would it be possible to merge this one before we merge any other PR related to STM32 board introduction/configuration?

File: boards/arm/disco_l475_iot1/disco_l475_iot1.yaml
Comment (galak) [inline boards/arm/disco_l475_iot1/disco_l475_iot1.yaml]: should this be ble, not bluetooth?

**Review by galak — CHANGES_REQUESTED** [review]
I'm not ok with the changes to build_all/testcase.yaml.  This limits / reduces greatly the number of platforms the build all tests are run on.  We need a different solution.  Also, I don't want to introduce even more things into the board.yaml for 'supported' by.  We should be getting that info from DT, rather than duplicating it in yet another place.

**Comment by erwango** [comment]
> I'm not ok with the changes to build_all/testcase.yaml. This limits / reduces greatly the number of platforms the build all tests are run on. We need a different solution. Also, I don't want to introduce even more things into the board.yaml for 'supported' by. We should be getting that info from DT, rather than duplicating it in yet another place.

@ydamigos : Did you found failed tests w/o this change and was that only pure cleanup?

**Comment by ydamigos** [comment]
> @ydamigos : Did you found failed tests w/o this change and was that only pure cleanup?

Without commits 7b6cf2c2e51d12537c238adb6ebd29ab4b8e265e and e61b3192abc3eb9230cb318d605652f2a15e5cd0 the build_all test cases failed on various STM32 boards.

**Comment by galak** [comment]
@ydamigos with the changes you mention the number of boards/platforms build_all test is greatly reduced.  I looked at just ethernet and we go from 40+ something building it to 5.

**Comment by erwango** [comment]
@galak: I agree we should try to find a way not reducing too much the number of tests done.

Though , about 
> Also, I don't want to introduce even more things into the board.yaml for 'supported' by. We should be getting that info from DT, rather than duplicating it in yet another place.

While I totally on the theory, can we decorrelate this topic from DT? I think it is important this PR is merged in short term (then before we have a **complete** DTS description for each board and DTS matching information extracted)

**Comment by galak** [comment]
> While I totally on the theory, can we decorrelate this topic from DT? I think it is important this PR is merged in short term (then before we have a complete DTS description for each board and DTS matching information extracted)

I just dont want to introduce new 'supported' by items like 'uart' that don't exist anywhere in the tree today (from what I can tell).

**Comment by ydamigos** [comment]
> I just dont want to introduce new 'supported' by items like 'uart' that don't exist anywhere in the tree today (from what I can tell).

xtensa/esp32/esp32.yaml:11 already contains 'uart'.

To be exact, only commit 7b6cf2c is needed to avoid build errors. An alternative solution might be to blacklist the boards that fail the tests. But every new STM32 board introduced should be also blacklisted in the tests if it does not use the controllers (spi, i2c, adc).

**Comment by galak** [comment]
How about in arch/arm/soc/st_stm32/common/Kconfig.defconfig.series we add:

if SPI && (SPI_1 || SPI_2 || SPI_3)

so that way we only enable the spi driver if a port is enabled as well.

**Comment by galak** [comment]
We'll need something similar to SPI for I2C ports

**Comment by ydamigos** [comment]
I will drop commits 7b6cf2c, e61b319 and I will add the extra checks in common/Kconfig.defconfig.series

**Comment by galak** [comment]
You might also need to reorder the commits.  I think the GPIO port fixes, might need to be first.

**Comment by galak** [comment]
Here's a diff as a starting point if it helps any - https://paste2.org/gGWXhWLk

**Comment by ydamigos** [comment]
>How about in arch/arm/soc/st_stm32/common/Kconfig.defconfig.series we add:
if SPI && (SPI_1 || SPI_2 || SPI_3)
so that way we only enable the spi driver if a port is enabled as well.

@galak Thanks for the solution. It is nice and clean.

**Review by galak — APPROVED** [review]

**Pull request merged**
